packages/affiliates/libs/affiliates.js

```diff
@@ -0,0 +1,192 @@
+const { EmpBalancesHistory } = require("./processors");
+const { Prices } = require("./models");
+const { DecodeLog } = require("./contracts");
+const highland = require("highland");
+const moment = require("moment");
+const Promise = require("bluebird");
+const assert = require("assert");
+const { parseFixed } = require("@uma/common");
+
+const DeployerRewards = ({ queries, empCreatorAbi, empAbi, coingecko }) => {
+  assert(coingecko, "requires coingecko api");
+  assert(queries, "requires queries class");
+  assert(empCreatorAbi, "requires creator abi");
+  assert(empAbi, "requires empAbi");
+  async function getBalanceHistory(address, start, end) {
+    // stream is a bit more optimal than waiting for entire query to return as array
+    const stream = await queries.streamLogsByContract(address, start, end);
+    const decode = DecodeLog(empAbi);
+    const balancesHistory = EmpBalancesHistory();
+    await highland(stream)
```

Comment:
What does this `highland()` function actually do and how does it relate `stream` to `balancesHistory`? Looks like `balanceHistory` is empty until you run `highland(stream)` which does something with `balanceHistory.handleEvent`. 

Comments might be helpful

Reply:
highland is to streams as lodash is to arrays.  It essentially wraps a native node stream (or iterator, event emitter, promise) into a stream with chainable utility functions.

It essentialy provides the .doto, .last .toPromise which native node streams dont have.